### ISSUE TITLE
TARS modell training doesn't proceed after first epoch (1000+ labels in dictionary)

### ISSUE THREAD

**Issue by MattesR** [flairNLP/flair#2495]
**Describe the bug**
I want to find out whether pre-training the TARS model for an argument aspect classification task with other data improves our scores. For that, I'm training with the [Webis Dataset](https://webis.de/data/webis-argument-framing-19.html), using their frames as classes. There are 1493 labels in the dictionary which is a lot. Is that a problem?

After the first epoch, the training just stops. Or does it take over half an hour to start the second epoch? I'd assume no

**To Reproduce**
Here's the Code I use for creating the dataset. My GPU has 8GB of RAM, after the first epoch, roughly 6 GB are allocated.

```
from flair.data import Corpus
from flair.datasets import CSVClassificationCorpus
from flair.models import TARSClassifier
from flair.trainers import ModelTrainer
import torch

label_type = 'question_class'

def get_corpus_from_path(path): # creating the corpus works
    column_name_map = {1: "text", 2: "label_topic"}
    return CSVClassificationCorpus(
        path,
        column_name_map,
        label_type=label_type,
        skip_header=True
    )

def train_model(path_tuple):
    torch.cuda.empty_cache()  # just for making sure nothing hogs memory
    tars = TARSClassifier.load("tars-base")
    corpus = get_corpus_from_path(path_tuple[0]) # path_tuple[0] just has the path
    label_dict = corpus.make_label_dictionary(label_type=label_type)
    tars.add_and_switch_to_new_task(
        f"WEBIS_{path_tuple[1]}", # path_tuple[1] contains a code for the test run
        label_dictionary=label_dict,
        label_type=label_type
    )
    trainer = ModelTrainer(tars, corpus)
    trainer.train(
        base_path=f'{OUTPUT_PATH}/models/{path_tuple[1]}', # path to store the model artifacts
        learning_rate=0.02, # use very small learning rate
        mini_batch_size=16,
        mini_batch_chunk_size=4,
        max_epochs=10, # terminate after 10 epochs
    )
```
The log ends with `2021-11-01 19:51:22,653 EPOCH 1 done: loss 0.0314 - lr 0.0200000`

The program is still running right now (20:10) but nothing happens.

**Expected behavior**
The model finishing training.

**Environment (please complete the following information):**
 - OS [e.g. iOS, Linux]: Linux Kubuntu 20.04
 - Version [e.g. flair-0.3.2]: flair==0.9

**Additional context**
Do you think pre-training with that many classes is useful? I would like to just see with the task at hand but alas, I don't get the pre-training working.

**Comment by alanakbik** [comment]
Hello @MattesR, cool dataset! The problem probably lies in the large number of labels: TARS predicting time scales linearly with the number of labels. Since we predict labels for the dev split at the end of each epoch, it becomes very slow here.

@kishaloyhalder has a PR in the works with a modified version of TARS that will be much faster in scenarios with many labels. (Could be a bit though until it gets added.)

**Comment by MattesR** [comment]
Thanks for the quick reply @alanakbik!
I had a hunch while waiting for the next epoch to start, that the number of labels might be problematic, since I had no problems training models with eight labels.
